tests/unit/source_code/linters/test_dfsa.py

```diff
@@ -0,0 +1,149 @@
+import pytest
+
+from databricks.labs.ucx.source_code.base import Deprecation, Advice, CurrentSessionState, Failure
+from databricks.labs.ucx.source_code.linters.dfsa import DfsaPyLinter, DfsaSqlLinter, DFSA_PATTERNS
+
+
+@pytest.mark.parametrize(
+    "path, matches",
+    [
+        ("/mnt/foo/bar", True),
+        ("dbfs:/mnt/foo/bar", True),
+        ("s3a://bucket1/folder1", True),
+        ("/dbfs/mnt/foo/bar", True),
+        ("/tmp/foo", False),
+        ("table.we.know.nothing.about", False),
+    ],
+)
+def test_matches_dfsa_pattern(path, matches):
+    """see https://github.com/databrickslabs/ucx/issues/2350"""
+    matched = any(pattern.matches(path) for pattern in DFSA_PATTERNS)
+    assert matches == matched
+
+
+@pytest.mark.parametrize(
+    "code, expected",
+    [
+        ('SOME_CONSTANT = "not a file system path"', 0),
+        ('SOME_CONSTANT = ("/dbfs/mnt", "dbfs:/", "/mnt/")', 3),
+        ('# "/dbfs/mnt"', 0),
+        ('SOME_CONSTANT = "/dbfs/mnt"', 1),
+        ('SOME_CONSTANT = "/dbfs/mnt"; load_data(SOME_CONSTANT)', 1),
+        ('SOME_CONSTANT = 42; load_data(SOME_CONSTANT)', 0),
+    ],
+)
+def test_detects_dfsa_paths(code, expected):
+    linter = DfsaPyLinter(CurrentSessionState(), allow_spark_duplicates=True)
+    advices = list(linter.lint(code))
+    for advice in advices:
+        assert isinstance(advice, Advice)
+    assert len(advices) == expected
+
+
+@pytest.mark.parametrize(
+    "code, expected",
+    [
+        ("load_data('/dbfs/mnt/data')", 1),
+        ("load_data('/data')", 1),
+        ("load_data('/dbfs/mnt/data', '/data')", 2),
+        ("# load_data('/dbfs/mnt/data', '/data')", 0),
+        ('spark.read.parquet("/mnt/foo/bar")', 1),
+        ('spark.read.parquet("dbfs:/mnt/foo/bar")', 1),
+        ('spark.read.parquet("dbfs://mnt/foo/bar")', 1),
+        ('DBFS="dbfs:/mnt/foo/bar"; spark.read.parquet(DBFS)', 1),
+        (
```

Comment:
use functional tests for examples of more than one line, otherwise it's unmaintainable

Reply:
done